wormhole-attester/program/src/attest.rs

```diff
@@ -240,6 +267,51 @@ pub fn attest(ctx: &ExecutionContext, accs: &mut Attest, data: AttestData) -> So
 
     trace!("Attestations successfully created");
 
+    // Serialize the state to calculate rent/account size adjustments
+    let serialized = accs.attestation_state.1.try_to_vec()?;
```

Comment:
I think I would make one PDA per price feed to avoid the complexity of resizing the account

Reply:
Also potentially might congest the network because write access from many transactions to one account?

Reply:
about congestion, i had the same doubt, but Wormhole Sequence Account is already a bottleneck. So it won't be different.

Reply:
Ok then it's just about the extra complexity of having a variable size account

Reply:
I think this is fine. We're probably not going to exceed the size of the one account, and PDAs have their own issues.

Honestly either way would probably work, but I don't think there's a strong reason to rewrite this to use PDAs.

Reply:
My main motivation for using a single account is a nasty edge case on Solana. In general, you're allowed to transfer SOL to any non-executable account, PDA or otherwise. For uninitialized accounts, such transfer instantiates the account and assigns ownership to the system program. When a program encounters such auto-initialized account as its own PDA, the program cannot change the contents, because it does not own it. This potentially creates a DoS condition for any known uninitialized PDA. This is completely permissionless and can happen on accident, which means that even Pythnet is not 100% immune. In fact, an accident is how I learned, some time ago while debugging a program I thought I was making a rent mistake so I tried pre-loading an account with SOL, locking the program out of it. 

The attack surface in this situation is dependent on how often new PDAs need to be created and how easy (or not) it is to use a different PDA for the desired transaction. If we resort to a per-symbol PDA scheme, each new oracle symbol can potentially be vulnerable. What is even worse, if one symbol is denied its PDA, the only possible fix is to change the seeds inside program logic, losing all on-chain state for all the other accounts. 

The proposed single-PDA solution collapses the attack surface into just a single race condition in the whole lifetime of this logic, along with relatively rare upgrades of the attestation state (we version this PDA's format in its seeds). Any lockout condition can be worked around by just altering its seeds.

Reply:
As for storage space, the account size on Solana is limited, but the current 10 MiB limit seems sufficient. E.g. for 10k symbols we could give a bit more than 1KiB space per symbol. IMO this is more symbols than we'll have in forseeable future, and more space than we'll need in forseeable future